File: src/at/fhooe/mc/emg/util/PeakDetector.java
package at.fhooe.mc.emg.util;

import java.util.List;

/**
 * Author:  Mescht
 * Date:    08.07.2017
 */
public class PeakDetector {


    public static int detectSimpleThresholdPeaks(double[] yVals, int threshold) {

        int peaks = 0;
        boolean isRising = true;
        for (double yVal : yVals) {

            if (yVal > threshold && isRising) {
                isRising = false;
            }
            if (yVal < threshold && !isRising) {
                peaks++;
                isRising = true;
            }
        }
        return peaks;
    }

    public static List<Integer> detectPeakLocations(double[] yVals) {
        return null;
    }

}

File: core/src/at/fhooe/mc/emg/core/util/PeakDetector.kt
package at.fhooe.mc.emg.core.util

import io.reactivex.Observable
import io.reactivex.schedulers.Schedulers

/**
 * Author:  Martin Macheiner
 * Date:    08.07.2017
 */
// TODO Implement peaks detector
object PeakDetector {

    fun detectSimpleThresholdPeaks(yVals: DoubleArray, threshold: Int): Int {

        var peaks = 0
        var isRising = true
        for (yVal in yVals) {

            if (yVal > threshold && isRising) {
                isRising = false
            }
            if (yVal < threshold && !isRising) {
                peaks++
                isRising = true
            }
        }
        return peaks
    }

    fun detectPeakLocations(data: DoubleArray): Observable<IntArray> {
        return Observable.defer {
            Observable.just(IntArray(data.size))
        }.subscribeOn(Schedulers.computation())
    }

}
